python/pyarrow/__init__.py

```diff
@@ -412,14 +412,24 @@ def append_library_dir(library_dir):
                 append_library_dir(library_dir[2:])
 
     if _sys.platform == 'win32':
+        python_base_install = _os.path.dirname(_sys.executable)
+
         # TODO(wesm): Is this necessary, or does setuptools within a conda
         # installation add Library\lib to the linker path for MSVC?
-        python_base_install = _os.path.dirname(_sys.executable)
         library_dir = _os.path.join(python_base_install, 'Library', 'lib')
 
         if _os.path.exists(_os.path.join(library_dir, 'arrow.lib')):
             append_library_dir(library_dir)
 
+        # GH-45530: Add pyarrow.libs dir containing delvewheel-mangled
+        # msvcp140.dll
+        pyarrow_libs_dir = _os.path.join(
+            python_base_install, "Lib", "site-packages", "pyarrow.libs"
+        )
```

Comment:
Hmm, no, this is inflexible. Packages are indeed often installed in `site-packages`, but they could be elsewhere on `PYTHONPATH`.

I think this should really be using the `pyarrow.libs` directory that is next to the current package's directory.

Reply:
Changed in https://github.com/apache/arrow/pull/45766/commits/9e2526c61884101d9663830b9e03aa5e49855b87. Testing now.